<?xml version="1.0"?> 
<launch>

<param name="/use_sim_time" value="true"/>

<node name="rviz" pkg="rviz" type="rviz" args="-d $(find stage_gym)/rviz/stage.rviz"/>

<node name="stage_gym" pkg="stage_gym" type="stageros" args="$(find stage_gym)/world/basic.world"  output="screen" required="true">
		<remap from="base_scan" to="scan" />
		<param name="step_by_step"               value="true"/>
<param name="orca_robots" value="7"/>
</node>
<group ns="robot_0">
<node name="amcl" pkg="amcl" type="amcl" output="screen">
<remap from="scan" to="ranger_0/base_scan"/>
<param name="use_map_topic"             value="true"/>
<param name="odom_frame_id" value="/robot_0/odom" />
<param name="base_frame_id" value="/robot_0/base_link" />
<param name="global_frame_id" value="/map" />
<remap from="static_map" to="/static_map"/>
<remap from="map" to="/map"/>
<!-- Initial position -->
<param name="initial_pose_x" value="-1.93252" />
<param name="initial_pose_y" value="1.17936" />
<param name="initial_pose_a" value="-1.90161" />
<!-- amcl specific -->
<param name="odom_model_type"           value="diff"/>
<param name="gui_publish_rate"          value="10.0"/>
<param name="laser_max_beams"           value="60"/>
<param name="laser_max_range"           value="12.0"/>
<param name="min_particles"             value="500"/>
<param name="max_particles"             value="2000"/>
<param name="kld_err"                   value="0.05"/>
<param name="kld_z"                     value="0.99"/>
<param name="odom_alpha1"               value="0.2"/>
<param name="odom_alpha2"               value="0.2"/>
<!-- translation std dev, m -->
<param name="odom_alpha3"               value="0.2"/>
<param name="odom_alpha4"               value="0.2"/>
<param name="odom_alpha5"               value="0.1"/>
<param name="laser_z_hit"               value="0.5"/>
<param name="laser_z_short"             value="0.05"/>
<param name="laser_z_max"               value="0.05"/>
<param name="laser_z_rand"              value="0.5"/>
<param name="laser_sigma_hit"           value="0.2"/>
<param name="laser_lambda_short"        value="0.1"/>
<param name="laser_model_type"          value="likelihood_field"/>
<param name="laser_likelihood_max_dist" value="2.0"/>
<param name="update_min_d"              value="0.1"/>
<param name="update_min_a"              value="0.2"/>
<param name="resample_interval"         value="1"/>
<param name="transform_tolerance"       value="1.0"/>
<param name="recovery_alpha_slow"       value="0.0001"/>
<param name="recovery_alpha_fast"       value="0.1"/>
</node>
</group>
<node name="map_server" pkg="map_server" type="map_server" args="$(find stage_gym)/world/simple.yaml" required="true"/>
</launch>
